tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -131,6 +131,47 @@ class BaseExampleGenExecutor(
   ExampleGen can override `GenerateExamplesByBeam`.
   """
 
+  def __init__(self):
+    super(BaseExampleGenExecutor, self).__init__()
+    self.input_signature = {}
+    self.output_signature = {}
+
+  def beam_io_signature(self, input_dict: Dict[Text, List[types.Artifact]],
+                        output_dict: Dict[Text, List[types.Artifact]],
+                        exec_properties: Dict[Text, Union[int, float, Text]]
+                        ) -> Tuple[Dict[Text, type], Dict[Text, type]]:
+    return self.input_signature, self.output_signature
+
+  def read_inputs(self, pipeline: beam.Pipeline,
+                  input_dict: Dict[Text, List[types.Artifact]],
+                  output_dict: Dict[Text, List[types.Artifact]],
+                  exec_properties: Dict[Text, Union[int, float, Text]]
+                  ) -> Dict[Text, beam.pvalue.PCollection]:
+    inputs = self.GenerateExamplesByBeam(pipeline, input_dict, exec_properties)
+    return inputs
```

Comment:
I don't think this is correct. I think ExampleGen should be modeled as a component that takes no inputs (because it is a "source" component). `GenerateExampleByBeam` should be placed in `run_component` instead.

Reply:
I placed `GenerateExampleByBeam` in `run_component` as suggested.